### PR TITLE
add libc 2.12+ functionality for structs.py

### PR DESCRIPTION
**Pull request by shenhavmor10**

### CODE REVIEW

File: pwndbg/heap/structs.py [hunk 1]
```diff
@@ -290,6 +291,66 @@ class c_malloc_state_2_26(Structure):
     ]
 
 
+class c_malloc_state_2_12(Structure):
+    """
+    This class represents malloc_state struct for GLIBC < 2.22 as a ctypes struct.
+
+    https://github.com/bminor/glibc/blob/glibc-2.23/malloc/malloc.c#L1686-L1724
```
Comment (disconnect3d) [inline pwndbg/heap/structs.py]: Is this whole docstring valid? It says GLIBC<2.22 but did you really check GLIBCs 2.12 up to 2.23 that we do support?

Also the link points to glibc-2.23 branch, so it seems it maybe is a copy-paste leftover from c_malloc_par_2_23 struct.

Please fix the docstring

File: pwndbg/heap/structs.py [hunk 2]
```diff
@@ -594,6 +657,63 @@ class c_malloc_par_2_23(Structure):
         ("sbrk_base", c_pvoid),
     ]
 
+class c_malloc_par_2_12(Structure):
+    """
+    This class represents the malloc_par struct for GLIBC < 2.15 as a ctypes struct.
```
Comment (disconnect3d) [inline pwndbg/heap/structs.py:665]: Just to be sure: how do we know 2.15 is correct here? If we really know it, can we link to diff or code and state what has changed?

File: pwndbg/heap/structs.py [hunk 3]
```diff
@@ -796,8 +916,10 @@ class MallocPar(CStruct2GDB):
         _c_struct = c_malloc_par_2_26
     elif pwndbg.glibc.get_version() >= (2, 24):
         _c_struct = c_malloc_par_2_24
-    else:
+    elif pwndbg.glibc.get_version() >= (2, 15):
```
Comment (disconnect3d) [inline pwndbg/heap/structs.py]: Nitpick: if we do this change, can we call to get_version() once and save it in a `_version` variable or sth?

File: pwndbg/heap/structs.py (hunk 2 above)
Comment (CptGibbon) [reply pwndbg/heap/structs.py:665]: Indeed that "no description provided" for this PR is killing me

File: pwndbg/heap/structs.py (hunk 1 above)
Comment (shenhavmor10) [reply pwndbg/heap/structs.py]: ye, But the link needs a fix :)

**Comment by shenhavmor10** [comment]
image
in glibc 2.15 the pagesize variable is removed.
image
here the attached threads variable is added in glibc 2.23

**Comment by disconnect3d** [comment]
Do we need to change comments for structs from 2.23 version which was previously the oldest one we supported?

E.g.
```py
class c_malloc_par_2_23(Structure):
    """
    This class represents the malloc_par struct for GLIBC < 2.24 as a ctypes struct.
```

I think this is currently not accurate bcoz its more like 2.15 <= GLIBC < 2.24?

**Pull request merged**